internal/engine/dcwatch/disable_watcher.go

```diff
@@ -26,64 +53,98 @@ func (w *DisableSubscriber) OnChange(ctx context.Context, st store.RStore, summa
 		return nil
 	}
 
-	if len(summary.UIResources.Changes) == 0 {
-		return nil
-	}
-
 	state := st.RLockState()
-	project := state.DockerComposeProject()
 
+	project := state.DockerComposeProject()
 	if model.IsEmptyDockerComposeProject(project) {
 		st.RUnlockState()
 		return nil
 	}
 
-	var uirToDisable *v1alpha1.UIResource
-	var specToDisable model.DockerComposeUpSpec
-	for nn := range summary.UIResources.Changes {
-		uir, ok := state.UIResources[nn.Name]
-		if !ok {
-			continue
-		}
-		manifest, exists := state.ManifestTargets[model.ManifestName(uir.Name)]
-		if !exists {
+	for i, mn := range state.ManifestDefinitionOrder {
+		mt := state.ManifestTargets[mn]
+		ms := mt.State
+
+		if !ms.IsDC() {
 			continue
 		}
-		ms := manifest.State
-		if manifest.State.DisableState == v1alpha1.DisableStateDisabled && ms.IsDC() {
-			rs := ms.DCRuntimeState().RuntimeStatus()
-			if rs == v1alpha1.RuntimeStatusOK || rs == v1alpha1.RuntimeStatusPending {
-				// for now, only disable one at a time
-				// https://app.shortcut.com/windmill/story/13140/support-logging-to-multiple-manifests
-				specToDisable = manifest.Manifest.DockerComposeTarget().Spec
-				uirToDisable = uir
-				break
-			}
+
+		runtimeStatus := ms.DCRuntimeState().RuntimeStatus()
+
+		isRunning := runtimeStatus == v1alpha1.RuntimeStatusOK || runtimeStatus == v1alpha1.RuntimeStatusPending
+		isDisabled := ms.DisableState == v1alpha1.DisableStateDisabled
+		needsCleanup := isRunning && isDisabled
+
+		rs := resourceState{
+			Spec:         mt.Manifest.DockerComposeTarget().Spec,
+			NeedsCleanup: needsCleanup,
+			StartTime:    ms.DCRuntimeState().StartTime,
+			Order:        i,
 		}
+		w.mu.Lock()
+		w.resourceStates[rs.Spec.Service] = rs
+		w.mu.Unlock()
 	}
 
 	st.RUnlockState()
 
-	if uirToDisable != nil {
-		// Upon disabling, the DC event watcher will notice the container has stopped and update
-		// the resource's RuntimeStatus, preventing it from being re-added to specsToDisable.
-		// There is a race here, since this OnChange might get called again before EngineState
-		// gets updated with the new RuntimeStatus. This is fine--calling `docker-compose rm` on
-		// a down service is a no-op.
-
-		ctx, err := store.WithObjectLogHandler(ctx, st, uirToDisable)
-		if err != nil {
-			logger.Get(ctx).Errorf("error creating logger for %s: %v", uirToDisable.Name, err)
-			// continue anyway:
-			// probably better to try stopping the service instead of re-logging error infinitely
-		}
-		out := logger.Get(ctx).Writer(logger.InfoLvl)
+	go func() {
```

Comment:
it would be good to do a quick check here and only schedule the goroutine if at least one service needs cleanup.

creating a new goroutine and Sleep() on every OnChange() is going to create problems for the goroutine scheduler; it's why the Store is careful about batching https://github.com/tilt-dev/tilt/blob/master/internal/store/subscriber.go#L124